app/Http/Controllers/ContactsController.php

```diff
@@ -740,11 +740,18 @@ public function list(Request $request)
         $perPage = $request->has('perPage') ? $request->input('perPage') : config('monica.number_of_contacts_pagination');
 
         // search contacts
-        $contacts = $contacts->search($request->input('search') ?? '', $accountId, 'is_starred', 'desc', $sort)
-            ->paginate($perPage);
+        $contacts = $contacts->search($request->input('search') ?? '', $accountId, 'is_starred', 'desc', $sort);
+
+        if($perPage == -1){
```

Comment:
I recommend creating separate methods. It might look like this:
```suggestion
        $total = $this->getTotal($contacts, $perPage);
        $contacts = $this->getContacts($contacts, $perPage);
```
Example of the `getTotal()` implementation:
```php
private function getTotal($contacts, int $perPage): int
{
    if ($perPage === -1) {
        return $contacts->count();
    }

    return $contacts->paginate($perPage);
}
```
Please add a type hint for the `$contacts` argument. The `getContacts()` method might look similar to the above.